### PR TITLE
UI - platform web initial form work

### PR DESCRIPTION
**Pull request by kialj876**
*Issue:*

- bcgov/entity/issues/

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namex license (Apache 2.0).

### CODE REVIEW

File: strr-pm-web/app/components/form/platform/Details.vue [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+                    :name="`brands.${i}.brandName`"
+                    :error="isComplete && !brand.name ? 'Required' : ''"
+                  >
+                    <ConnectField
+                      :id="'platform-brand-name-' + i"
+                      v-model="brand.name"
+                      :help="tPlat('hint.brandName')"
+                      :name="`brands.element.test`"
+                      :placeholder="(i > 0) ? tPlat('label.brandNameOpt') : tPlat('label.brandName')"
+                      :wrap-with-form-grp="false"
```
Comment (deetz99) [inline strr-pm-web/app/components/form/platform/Details.vue:64]: will the help text still be rendered when wrap-with-form-grp=false?

File: strr-pm-web/app/components/form/platform/Details.vue [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+                    :name="`brands.${i}.website`"
+                    :error="isComplete && !brand.website ? 'Required' : ''"
+                  >
+                    <ConnectField
+                      :id="'platform-brand-site-' + i"
+                      v-model="brand.website"
+                      :help="tPlat('hint.brandSite')"
+                      name="website"
+                      :placeholder="(i > 0) ? tPlat('label.brandSiteOpt') : tPlat('label.brandSite')"
+                      :wrap-with-form-grp="false"
```
Comment (deetz99) [inline strr-pm-web/app/components/form/platform/Details.vue:78]: will the help text still be rendered when wrap-with-form-grp=false?

File: strr-pm-web/app/composables/useScreenSize.ts
Comment (deetz99) [inline strr-pm-web/app/composables/useScreenSize.ts:1]: switch to vueuse? https://vueuse.org/core/useWindowSize/

**Comment by deetz99** [comment]
I'm happy to merge this now just commenting to keep track of things.

Looks pretty good just some small things like i18n/aria-labels, styling

- form/common/crainfo aria labels need i18n
- form/common/ReviewRow should the divs still be w-1/3 when flex direction = flex-col
- form/host/PrincipalResidence most aria-labels need i18n
- form/host/property need aria labels on ConnectSection components and aria-label i18n
- form/host/ReviewConfirm img alt text needs i18n, 'Undeclared' needs i18n
- form/platform/Details help text wont be rendered on ConnectField components when wrap with form grp prop = false
- useWindowSize from vueuse instead of custom useScreenSize composable
- missing i18n in stores/hostProperty
- fee widget, increase z-index (button icon in buttonControl button is showing through the fee widget heading when fee widget is collapsed on small screen), make heading focusable (and styling) when floating at the bottom on small screen, remove focus option when on large screen, currently fee widget covers the footer on small screens, maybe we could move the fee widget to be above the button control component on small screens? will need to think about this one)
- stepping through the from when a form is invalid causes a uncaught form validation error in the console
- something is causing the screen width to overflow under ~1370px (fee widget width?)
- t('createAccount.propertyForm.platformUrl') 'platform' typo

**Review by deetz99 — APPROVED** [review]

**Comment by kialj876** [comment]
> I'm happy to merge this now just commenting to keep track of things.
> 
> Looks pretty good just some small things like i18n/aria-labels, styling
> 
> * form/common/crainfo aria labels need i18n
> * form/common/ReviewRow should the divs still be w-1/3 when flex direction = flex-col
> * form/host/PrincipalResidence most aria-labels need i18n
> * form/host/property need aria labels on ConnectSection components and aria-label i18n
> * form/host/ReviewConfirm img alt text needs i18n, 'Undeclared' needs i18n
> * form/platform/Details help text wont be rendered on ConnectField components when wrap with form grp prop = false
> * useWindowSize from vueuse instead of custom useScreenSize composable
> * missing i18n in stores/hostProperty
> * fee widget, increase z-index (button icon in buttonControl button is showing through the fee widget heading when fee widget is collapsed on small screen), make heading focusable (and styling) when floating at the bottom on small screen, remove focus option when on large screen, currently fee widget covers the footer on small screens, maybe we could move the fee widget to be above the button control component on small screens? will need to think about this one)
> * stepping through the from when a form is invalid causes an uncaught form validation error in the console
> * something is causing the screen width to overflow under ~1370px (fee widget width?)
> * t('createAccount.propertyForm.platformUrl') 'platform' typo
> * form padding/margin is uneven on small screens

Yeah sounds good nice comments, I'll try to get most of these and then merge later today so that its easier for you to get started. I'll be leaving all the 'property manager / host' stuff alone since we're not doing it now (maybe we'll go back to it later)

File: strr-pm-web/app/components/form/platform/Details.vue (hunk 2 above)
Comment (kialj876) [reply strr-pm-web/app/components/form/platform/Details.vue:78]: reworked this a bit

**Pull request merged**